plugins/bcc-login/includes/class-bcc-login-feed.php

```diff
@@ -108,111 +108,86 @@ function add_internal_category_to_member_items($the_list, $type) {
-            if (!empty($post_groups))
+            // Make sure posts with a group set don't have public visibility
+            if ( !empty($post_target_groups) || !empty($post_visibility_groups) )
             {
                 if ($visibility == BCC_Login_Visibility::VISIBILITY_PUBLIC) {
                     $visibility = BCC_Login_Visibility::VISIBILITY_SUBSCRIBER;
                 }
             }
-
         }
-        $this->add_visibility_to_items($the_list,$visibility);
-        $this->add_groups_to_items($the_list, $visibility);
 
+        $this->add_visibility_to_items($the_list, $visibility);
+        $this->add_groups_to_items($the_list, $visibility);
     }
 
     // Include basic visibility settings in feed: 
     // - public (no authentication required)
     // - user (requires authentication)
     // - internal:{district name} (requires affiliation with organization in specified district)
-    function add_visibility_to_items ($the_list, $visibility) {
-        
-        if ($visibility == BCC_Login_Visibility::VISIBILITY_PUBLIC){
+    function add_visibility_to_items($the_list, $visibility) {
+        if ($visibility == BCC_Login_Visibility::VISIBILITY_PUBLIC) {
             echo "<bcc:visibility>public</bcc:visibility>\n"; 
-
-        } else if ($visibility == BCC_Login_Visibility::VISIBILITY_SUBSCRIBER){
+        } else if ($visibility == BCC_Login_Visibility::VISIBILITY_SUBSCRIBER) {
             echo "<bcc:visibility>user</bcc:visibility>\n"; 
-        } else if ($visibility == BCC_Login_Visibility::VISIBILITY_MEMBER){
+        } else if ($visibility == BCC_Login_Visibility::VISIBILITY_MEMBER) {
             echo "<bcc:visibility>internal:" . $this->_settings->member_organization_name . "</bcc:visibility>\n"; 
         }
     }
-
     
-    // Include group uid for each group that the post is visible for or targetted at (notification group)
-    // E.g. 
-    // <bcc:visiblityGroup>d4c434a7-504a-4246-9a10-def7dbfa982c</bcc:visiblityGroup>
-    // <bcc:visiblityGroup>25f5bc4d-48e0-4a6e-bf05-6b2a15d70861</bcc:visiblityGroup>
-    // <bcc:notificationGroup>d4c434a7-504a-4246-9a10-def7dbfa982c</bcc:notificationGroup>
+    // Include group uid for each group that the post is targeted at or visible for
+    // E.g.:
+    // <bcc:targetGroup>d4c434a7-504a-4246-9a10-def7dbfa982c</bcc:targetGroup>
+    // <bcc:visibilityGroup>d4c434a7-504a-4246-9a10-def7dbfa982c</bcc:visibilityGroup>
+    // <bcc:visibilityGroup>25f5bc4d-48e0-4a6e-bf05-6b2a15d70861</bcc:visibilityGroup>
     function add_groups_to_items($the_list, $visibility) {
         global $post;
         $result = '';
+
         if ( !empty($this->_settings->site_groups) || !empty($this->_settings->full_content_access_groups) ) {
-            // Get groups that are checked on post
-            $post_groups = get_post_meta($post->ID, 'bcc_groups', false);
+            $post_target_groups = get_post_meta($post->ID, 'bcc_groups', false);
```

Comment:
Would it be an idea to use bcc_target_groups instead of bcc_groups (for clarity), but still include bcc_groups for backwards compatibility?